js/lex.go

```diff
@@ -66,21 +90,34 @@ func (tt TokenType) String() string {
 
 // Lexer is the state for the lexer.
 type Lexer struct {
-	r *buffer.Lexer
-
-	regexpState   bool
-	templateState bool
+	r     *buffer.Lexer
+	stack []ParsingContext
```

Comment:
Then it might be confusing given that this is tokenizer otherwise. "Parsing context" means it actually stores context as proper parser would on top of tokenizer, except does that in a much simpler way without full parser available.